solution/ui/prototype/src/views/Part.vue

```diff
@@ -147,31 +276,96 @@ export default {
         tabsContent() {
             return [this.tocContent, this.partContent, null, null];
         },
-        contentContainerResourcesClass() {
-            return `content-container-${this.resourcesDisplay}`;
-        },
     },
 
     async created() {
-        this.$watch(
-            () => this.$route.params,
-            (toParams, previousParams) => {
-                // react to route changes...
-                if (toParams.part !== previousParams.part) {
-                    this.structure = null;
-                    this.clearResourcesParams();
-                    this.title = toParams.title;
-                    this.part = toParams.part;
-                    this.resourcesDisplay =
-                        toParams.resourcesDisplay || "drawer";
-                }
+        for (const key of Object.keys(this.queryParams)) {
+            if (key == "subpart") {
+                this.tabLabels[key] = this.formatTabLabel(key);
+            } else if (key == "section") {
+                this.tabLabels[key] = this.formatTabLabel(key);
             }
+        }
+        await this.getPartStructure();
+        await this.getFormattedSubpartsList(this.part);
+        await this.getFormattedSectionsList(
+            this.part,
+            this.queryParams.subpart
         );
 
-        await this.getPartStructure();
+        if (_isEmpty(this.queryParams)) {
+            let paramsToSet = {};
+            if (this.tabParam == "subpart") {
+                paramsToSet = {
+                    subpart: this.tabsShape.subpart.listItems[0].identifier,
+                };
+            }
+            if (this.tabParam == "section") {
+                paramsToSet = {
+                    subpart: this.tabsShape.subpart.listItems[0].identifier,
+                    section: this.tabsShape.section.listItems[0].identifier,
+                };
+            }
+            this.$router.push({
+                name: "part",
+                params: {
+                    title: this.title,
+                    part: this.part,
+                    tab: this.tabParam,
+                },
+                query: paramsToSet,
+            });
+        }
     },
 
     methods: {
+        formatTabLabel(type) {
```

Comment:
I made a comment up above about this, maybe instead just add a default no op option to this switch statement.